src/js/Controllers/TTSController.js

```diff
@@ -14,8 +15,10 @@ class TTSController {
         this.listener = listener
     }
 
-    onResetTimeout(appID, methodName) {
-        this.listener.send(RpcFactory.OnResetTimeout(appID, "TTS", methodName))
+    onResetTimeout(messageId) {
+        let resPeriod = store.getState().system.resetPeriod;
+
+        this.listener.send(RpcFactory.OnResetTimeout(messageId, 'TTS.Speak', resPeriod));
```

Comment:
```suggestion
        this.listener.send(RpcFactory.OnResetTimeout(messageId, 'TTS.Speak', 10000));
```

I don't think this value needs to be configurable, it just needs to be `>=` the value in `setInterval`. Currently if you make the value anything less than 9000 issues will arise. But I don't see a need to expose changing this value...

If you agree, we can remove the `resetPeriodSelector` in index.js and the corresponding action and reducer. Otherwise, please explain why this is kept and adjust the parameter of `setInterval` accordingly.

Reply:
@iCollin thank you for your comments! In case if we remove "resetPeriodSelector in index.js and the corresponding action and reducer" we will not have possibility to test it any more. Value of resetPeriod is one value for all RPC resetting as well. If we don't need it, then could you help to understand what was the purpose of [proposal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md)?

Reply:
This statement is true: ```HMI would be responsible to fine tune the wait time per method call as needed. It is up to HMI to control number of reset timeouts and duration of each reset timeout for endless or finite method timeout.```, but it does not mean the user should control the timeouts. It means that the `resetPeriod` is now a parameter of the `OnResetTimeout` RPC instead of being hardcoded in Core.

Reply:
@iCollin thank you very much for details! I removed resetPeriodSelector and all the corresponding. Please find the changes [here](https://github.com/LuxoftSDL/generic_hmi/pull/39/files).